### PR TITLE
Create database

### PR DESCRIPTION
**Pull request by BrookJeynes**
## Notes
- Created database connection via connection pools
- Created routes
    - api/review
    - api/review/:id
    - api/review/user/:id
    - api/review/create
    - api/review/approve/:id
    - api/review/:id
    - api/review/:id
    - api/unit/create
    - api/user/create

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -8,7 +8,25 @@
...
+
+# Whether the api returns dummy data or not
+DEMO=false 
+DATABASE_URL=postgres://username:password@localhost/qut_review
+```
+
+## Setting up your local database
+
+1. Install PostgreSQL
+2. Create the database: `CREATE DATABASE qut_review`
```
Comment (rjboas) [inline api/README.md:29]: Add a note that this can be done from the postgre CLI with `createdb` perhaps?

File: api/README.md [hunk 2]
```diff
@@ -8,7 +8,25 @@
...
```
Comment (rjboas) [inline api/README.md:30]: Add a note to install `diesel_cli` and run `migration run` in the infrastructure folder

File: api/api/src/review_handler.rs [hunk 3]
```diff
@@ -20,85 +25,83 @@ pub fn list_reviews_handler(demo_mode: Result<DummyResponse, NetworkResponse>) -
     serde_json::to_string(&response).expect("Return 500 internal server error.")
 }
 
-/// Takes in a `review_id` and returns a 200 OK with the associated review as JSON
-/// otherwise, a 404 NotFound NetworkResponse.
 #[get("/<review_id>")]
 pub fn list_review_handler(
-    review_id: u32,
+    review_id: i32,
```
Comment (rjboas) [inline api/api/src/review_handler.rs:30]: Why is review_id an `i32` now? Is there a reason it should be signed?

**Review by rjboas — COMMENTED** [review]
Had a look at the code and got the backend running on my system :+1: Just got those couple comments but otherwise looking good.

File: api/api/src/review_handler.rs (hunk 3 above)
Comment (BrookJeynes) [reply api/api/src/review_handler.rs:30]: Diesel doesn't seem to want to put anything other than an i32 into the database.

Comment (rjboas) [reply api/api/src/review_handler.rs:30]: Looks like postgresql doesn't support unsigned integers? I guess this is fine

File: api/README.md (hunk 1 above)
Comment (BrookJeynes) [reply api/README.md:29]: Will be leaving it as it's the most broad way of setting it up

File: api/README.md (hunk 2 above)
Comment (BrookJeynes) [reply api/README.md:30]: Added in another PR

**Review by rjboas — APPROVED** [review]

**Review by Tarang74 — COMMENTED** [review]
Happy with the changes as discussed yesterday

**Pull request merged**